trame/layouts/core.py

```diff
@@ -90,14 +109,12 @@ def start(self, port=None, debug=False):
         """
         _app = tri.get_app_instance()
 
-        _app.name = self.name
-        _app.layout = self.html
-        _app.on_ready = tri.print_server_info()
+        self._init_app(_app)
 
-        if self.favicon:
-            _app.favicon = self.favicon
         if self.on_ready:
             _app.on_ready = tri.print_server_info(self.on_ready)
+        else:
+            _app.on_ready = tri.print_server_info()
```

Comment:
Don't you want to handle the `_app.on_ready` part within your `self._init_app(_app)` as well?

Reply:
Actually, scratch that, I see below why you don't. ;-)